### ISSUE TITLE
Do not require a mod update for translations I do not use.

### ISSUE THREAD

**Issue by krs0** [vcmi/vcmi#4659]
**Describe your proposal**
Mods need to be updated for each translation update. Could they be excluded from update check? (maybe based on selected language?)

If a mod a few translations and work is done incrementally on those ... its a lot of unnecessary updates.
For ToW for example I had to do lately 3-4 35MB-ish updates (and opponent as well) just because of translations, that do not impact me at all.
HotA is 135MB...

**Comment by IvanSavenko** [comment]
Suggestions? Especially those that won't take month or two to implement?

> And if the compilations use different versions of a mod

How? Only one (latest) version may exists in repository. And you are not required to have latest version of a mod to play the game - you can play even if there is available update.

**Comment by dydzio0614** [comment]
I would say it is low priority for mow - compared to bugs and missing features that make people unable to fully enjoy VCMI without them and there are no workarounds, this issue is "first world problem" IMO

**Comment by krs0** [comment]
Yup definitely not a prio, is more of a "please keep this in mind if you change something in mod structure at some point". Somewhat related to "downloading the whole mod when only a submod changes"

...I do not know the system well enough for a thorough solution. What comes to mind:

Add version information to each Translation.
And instead of checking only parent mod, do a check on all submods and only! on enabled translation. 

But this will imply that modders will have to consistently update each submod info when those change, which is not a given for mods under heavy construction.

**Comment by krs0** [comment]
> Suggestions? Especially those that won't take month or two to implement?
> 
> > And if the compilations use different versions of a mod
> 
> How? Only one (latest) version may exists in repository. And you are not required to have latest version of a mod to play the game - you can play even if there is available update.

I could not update a submod in Launcher and wrongly drew the conclusion that it depends on a fixed version :(. But yes in "depends" : section there are no versions specified.

This raises another possible issue: A mod compilation will automatically always have to use the latest versions of its dependencies, if a conflict appears the compilation will simply not work anymore until a fix is provided.

![image](https://github.com/user-attachments/assets/d1566a21-a189-4551-b4a8-367ea39831e9)
